File: 02 - Even Fibonacci Numbers/Python/EvenFibonacciNumbers.py
# Project Euler - Problem 2
def fibonacci(max = float('inf')):
	prev = 1
	curr = 1
	while curr < max:
		yield curr
		prev, curr = curr, prev + curr

def main():
	sum = 0
	for x in fibonacci(4000000):
		sum += 0 if x % 2 else x

	print "Sum: ", sum

if __name__ == "__main__":
	main()

File: 02 - Even Fibonacci Numbers/Scala/EvenFibonacciNumbers.scala
// Project Euler - Problem 2
// Alex Tarrell

import Stream._

object EvenFibonacciNumbers {
	val fibonacci: Stream[Int] = 0 #:: fibonacci.scanLeft(1)(_ + _)

	def main(args: Array[String]) {
		println(fibonacci.filter(_ % 2 == 0).takeWhile(_ <= 4000000).sum)
	}
}